core/src/main/scala/org/http4s/multipart/MultipartParser.scala

```diff
@@ -648,33 +700,215 @@ object MultipartParser {
     }
   }
 
-  private def splitLimited[F[_]: Sync](
+  ////////////////////////////////////////////////////////////
+  // File writing encoder
+  ///////////////////////////////////////////////////////////
+
+  /** Same as the other streamed parsing, except
+    * after a particular size, it buffers on a File.
+    * @param maxSizeBeforeWrite Defaults to 50 megabytes
+    * @tparam F
+    * @return
+    */
+  def parseStreamedFile[F[_]: Sync](
+      boundary: Boundary,
+      limit: Int = 1024,
+      maxSizeBeforeWrite: Int = 52428800): Pipe[F, Byte, Multipart[F]] = { st =>
```

Comment:
So currently this helps in the common case that parts themselves are large. It doesn't help for a multipart of `N` parts which add up to the size of a large part. 

I am considering adding a configurable part limit to avoid abuse. Thoughts?